src/vm/analysis/type_checker/natives/maps.rs

```diff
@@ -38,9 +38,13 @@ pub fn check_special_fetch_entry(checker: &mut TypeChecker, args: &[SymbolicExpr
 pub fn check_special_fetch_contract_entry(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
     check_arguments_at_least(3, args)?;
     
-    let contract_name = args[0].match_atom()
-        .ok_or(CheckErrors::ContractCallExpectName)?;
-    
+
+    let contract_identifier = match args[0].expr {
```

Comment:
Same point here -- contract_identifiers for `contract-map-get` and `contract-call!` should always be literals.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1105/commits/c871057b86ad7ab2c64a84ba2986f42b469860ec